russstation/code/game/objects/items/miscellaneous.dm

```diff
@@ -0,0 +1,172 @@
+
+
+//end of construction code
+
+//Wetmore Slippery Sign
+//Constructable by janitors with upgrades from RND and sensors from cargo/robotics
+//Slips people who don't heed the warning
+
+//Wetmore Slippery Sign (traitor item)
+//purchasable from traitor uplink by janitors (7tc for 4)
+//in addition to slipping, also animates and kicks people around
+
+/obj/item/clothing/suit/caution/slippery
+	slot_flags = null //cannot be worn- or else it breaks stuff
+
+	var/willSlip = 0 //0 - disabled || 1 - enabled 
+	var/lastSlip = 0 //last time the sign slipped someone (time)
+	var/slipCooldown = 50 //cooldown (~5 seconds default)
+	var/clowningAround = 0 //0 - untainted || 1 - clown got it
+	var/evilSign = 0 //is the sign a traitor?
+	var/mob/living/carbon/boss = null //used so animated signs don't attack the janitor
+
+/obj/item/clothing/suit/caution/slippery/syndicate
+	evilSign = 1 //signs purchased from the uplink
+
+/obj/item/clothing/suit/caution/slippery/examine(mob/user)
+	. = ..()
+	if(isobserver(user) || user.mind.assigned_role == "Janitor") //janitors and ghosts can see that it's a slippery sign
+		. += "<span class='notice'>This [src.name] is outfitted with an experimental sprayer.</span>"
+		if(clowningAround || evilSign)
+			. += "<span class='warning'>It's been tampered with.</span>"
+	else if(HAS_TRAIT(user, TRAIT_CLUMSY)) //clowns know lube when they see it
+		. += "<span class='notice'>This [src.name] has great potential for pranks.</span>"
+
+
+//when used by a janitor: toggles between active and disabled, when used by a clown, pranks ensue
+/obj/item/clothing/suit/caution/slippery/attack_self(mob/user)
+	if(!proximity_monitor)
+		proximity_monitor = new(src, 1) //initializes the proximity: (source, range)
+
+	if(user.mind.assigned_role == "Janitor") //only janitors can interact with it normally
+		boss = user
+		willSlip = !willSlip
+		if(clowningAround) //so you can reset the sign if a clown messes with it
+			clowningAround = 0
+			slipCooldown = 50 
+
+		if(willSlip)
+			to_chat(user, "<span class='notice'>The [src.name] will now slip anyone running past.<span>")
+		else
+			to_chat(user, "<span class='notice'>The [src.name] will no longer slip passerbys.<span>")
+	else if(HAS_TRAIT(user, TRAIT_CLUMSY)) //clowns.
+		willSlip = 1 
+		clowningAround = 1
+		slipCooldown = 0
+		to_chat(user, "<span class='warning'>The [src.name]'s lube sprayer has been overloaded.<span>")
+	else
+		to_chat(user, "<span class = 'notice'>The [src.name] requires a janitor to activate.<span>")
+
+/obj/item/clothing/suit/caution/slippery/HasProximity(atom/movable/AM)
+	if (world.time < lastSlip + slipCooldown && lastSlip) //cooldown for slipping
+		return
+
+	if(!willSlip) //needs to be enabled to slip people obviously
+		return
+
+	if(!istype(AM, /mob/living/carbon)) //is it actually a thing we can slip?
+		return
+	
+	var/mob/living/carbon/C = AM
+	var/turf/open/T = get_turf(src)
+	var/list/adjacent_T = get_adjacent_open_turfs(T)
```

Comment:
Do we need to store this in a variable if it's only being used once, in an already fairly short line? Also on a stylistic note, you're combining `snake_case` with `camelCase` with that capital T. `snake_case` is generally more idiomatic in TGcode.

Reply:
which of the variables are you referring to? If I assume the carbon variable, the if statement doesn't want to function if using **AM** instead of **C**

Reply:
`adjacent_T`, instead of `adjacent_t` or `adjacentT`

Reply:
Oh wait, you meant for the one that's only being used once. Either way the answer is the same, `adjacent_T`